packages/apps/tools/src/components/Global/CustomTree/CustomTree.tsx

```diff
@@ -0,0 +1,224 @@
+      {(item) => {
+        return (
+          <>
+            <Stack
+              alignItems={'center'}
+              justifyContent={'space-between'}
+              onClick={() => {
+                item.toggleExpandCollapse();
+
+                // We "toggle" the expanded state ourselves since this is still the previous state
+                onExpandCollapse(item.data, !item.isExpanded);
+              }}
+              role="button"
+              className={itemContainerStyle}
+            >
+              <Button
+                variant="transparent"
+                onClick={() => {
+                  item.toggleExpandCollapse();
+
+                  // We "toggle" the expanded state ourselves since this is still the previous state
+                  onExpandCollapse(item.data, !item.isExpanded);
+                }}
+              >
+                {item.isExpanded ? <MonoArrowDropDown /> : <MonoArrowRight />}
+              </Button>
+              <Text className={itemTitleStyle} bold>
+                {item.data.title}
+              </Text>
+              {item.data.supportsReload && !item.data.isLoading ? (
+                <Button
+                  variant="transparent"
+                  isCompact
+                  onPress={() => {
+                    onReload(item.data);
+                  }}
+                  className={reloadButtonStyles}
+                >
+                  <MonoCached
+                    color={token('color.icon.semantic.positive.default')}
+                  />
+                </Button>
+              ) : null}
+              {!item.data.isLoading && item.data.label ? (
+                <Badge size="sm" style={'highContrast'}>
+                  {item.data.label}
+                </Badge>
+              ) : null}
+              {item.data.isLoading ? (
+                <Badge size="sm" style={'highContrast'}>
+                  Loading
+                </Badge>
+              ) : null}
+            </Stack>
+            {item.isExpanded ? (
+              <Node
+                data={item.data.children}
+                level={1}
+                onItemClick={onItemClick}
+                onExpandCollapse={onExpandCollapse}
+              />
+            ) : null}
+          </>
+        );
+      }}
+    </CustomAccordion>
+  );
+}
+
+interface INodeProps<T>
+  extends Omit<ICustomAccordionProps<T>, 'data' | 'children'> {
+  data: TreeItem<T>[];
+  level: number;
+  onItemClick: (item: TreeItem<T>) => void;
+  onExpandCollapse: (item: TreeItem<T>, expanded: boolean) => void;
+}
+
+// eslint-disable-next-line react/function-component-definition
+function Node<T>({
+  data,
+  level,
+  onItemClick,
+  onExpandCollapse,
+  ...rest
+}: INodeProps<T>) {
+  return (
+    <CustomAccordion
+      {...rest}
+      data={data}
+      itemProps={{
+        style: {
+          borderBlockEnd:
+            level === 1 ? '1px solid rgba(0, 0, 0, 0.25)' : 'initial',
+        },
```

Comment:
I would suggest to not use inline styles, also you are not using tokens for this color. So moving this to a css file is a win win for that.

Reply:
You could also opt in for a optional prop on the item. It's a custom component build for the use cases in the tools app..